qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,213 @@
+            qemu_image.check_image(params, data_dir)
+        params = self.params.object_params(self.source_image)
+        if compare_image:
+            compare_image = utils_misc.get_path(data_dir, compare_image)
+            source_image = qemu_storage.storage.get_image_filename(params, data_dir)
+            qemu_image.compare_images(compare_image, source_image)
+
+    def reopen(self):
+        """
+        Closing the vm and reboot it with the backup image.
+        """
+        vm = self.vm
+        vm.destroy()
+        image_chain = self.image_chain
+        image_name = self.params.get("image_name_%s" %
+                                     image_chain[len(image_chain) - 1])
+        self.params["image_name_%s" % self.source_image] = image_name
+        vm.create(params=self.params)
+        self.vm = vm
+        session = vm.wait_for_login(
+            timeout=int(self.params.get("login_timeout", 360)))
+        session.close()
+
+    def before_full_backup(self):
+        """
+        Run steps before create full backup.
+        """
+        return self.do_steps("before_full_backup")
+
+    def before_incremental(self):
+        """
+        Run steps before create incremental backup.
+        """
+        return self.do_steps("before_incremental")
+
+    def after_full_backup(self):
+        """
+        Run steps after create full backup.
+        """
+        self.do_steps("after_full_backup")
+        time.sleep(120)
+
+    def after_incremental(self):
+        """
+        Run steps after create incremental backup.
+        """
+        return self.do_steps("after_incremental")
+
+    def verify_job_info(self):
+        """
+        Check block job settings correctness.
+        """
+        params = self.params
+        device_id = self.get_device()
+        info = self.vm.monitor.info_block().get(device_id)
+        check_params = params.get("check_params")
+        for check_param in check_params:
+            if check_param in params:
+                target_value = params[check_param]
+                if check_param == "granularity":
+                    dirty_bitmap = info.get("dirty-bitmaps", "0")
+                    value = int(dirty_bitmap[0].get("granularity", "0"))
+                else:
+                    value = info.get(check_param, "")
+                if value != target_value:
+                    raise error.TestFail(
+                        "%s unmatched. Target is %d, result is %d" %
+                        (check_param, target_value, value))
+
+    def create_files(self):
+        """
+        Create files and record m5 values of them.
+        """
+        file_names = self.params["file_names"]
+        for file_name in file_names.split():
+            self.create_file(file_name)
+
+    def verify_md5s(self):
+        """
+        Check if the md5 values matches the record ones.
+        """
+        file_names = self.params["file_names"]
+        for file_name in file_names.split():
```

Comment:
@xutian , thanks, it will be reported inside self.verify_md5 function, in block_copy.py